Content.Server/MachineLinking/System/SignalLinkerSystem.cs

```diff
@@ -1,293 +1,338 @@
-using System.Collections.Generic;
 using System.Linq;
-using Content.Server.Hands.Components;
-using Content.Server.Interaction;
+using System.Diagnostics.CodeAnalysis;
 using Content.Server.MachineLinking.Components;
 using Content.Server.MachineLinking.Events;
-using Content.Server.MachineLinking.Exceptions;
-using Content.Server.MachineLinking.Models;
 using Content.Server.Power.Components;
-using Content.Server.UserInterface;
 using Content.Shared.Interaction;
 using Content.Shared.MachineLinking;
 using Content.Shared.Popups;
 using Robust.Server.GameObjects;
-using Robust.Shared.GameObjects;
-using Robust.Shared.Localization;
 using Robust.Shared.Utility;
+using Robust.Shared.Player;
 
 namespace Content.Server.MachineLinking.System
 {
     public sealed class SignalLinkerSystem : EntitySystem
     {
-        private InteractionSystem _interaction = default!;
+        [Dependency] private readonly UserInterfaceSystem _userInterfaceSystem = default!;
+        [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
 
-        private SignalLinkCollection _linkCollection = new();
+        private static readonly (string, string)[][] _defaultMappings =
+        {
+            new [] { ("Pressed", "Toggle") },
+            new [] { ("On", "On"), ("Off", "Off") },
+            new [] { ("On", "Open"), ("Off", "Close") },
+            new [] { ("On", "Forward"), ("Off", "Off") },
+            new [] { ("Left", "On"), ("Right", "On"), ("Middle", "Off") },
+            new [] { ("Left", "Open"), ("Right", "Open"), ("Middle", "Close") },
+            new [] { ("Left", "Forward"), ("Right", "Reverse"), ("Middle", "Off") },
+        };
```

Comment:
I'm still not sure about defining the default links in a giant list, rather than per component/port, but I guess this is fine for now,